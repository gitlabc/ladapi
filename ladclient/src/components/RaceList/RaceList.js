import styles from './styles.css';

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import RaceListItem from '../RaceListItem';

class RaceList extends Component {
    render() {
        const { top5Events, onRaceClick} = this.props;
        let now = Math.round((new Date()).getTime() / 1000);
        return (
            <div className={styles.rl}>
                {
                    top5Events.map((event) => {
                        return (
                            <RaceListItem key={event.id} event={event} now={now} onRaceClick={onRaceClick} />
                        )
                    }, this)
                }
            </div>
        );
    }
}

RaceList.propTypes = {
    top5Events: PropTypes.array.isRequired,
    onRaceClick: PropTypes.func.isRequired,
};

export default RaceList;