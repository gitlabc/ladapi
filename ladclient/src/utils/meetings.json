{
    "status": "Successful",
    "updates": [
        {
            "type": "next5",
            "data": [
                {
                    "id": 2589,
                    "name": "New Cyrus",
                    "type": "T",
                    "date": "2017-07-11",
                    "country": "AU",
                    "events": [
                        {
                            "id": 30395,
                            "race_num": 1,
                            "description": "Jimmie Islands",
                            "outcome": 1499722976,
                            "status": "open"
                        },
                        {
                            "id": 30589,
                            "race_num": 2,
                            "description": "Wuckert Keys",
                            "outcome": 1499723175,
                            "status": "open"
                        },
                        {
                            "id": 24203,
                            "race_num": 3,
                            "description": "Mohr Plain",
                            "outcome": 1499723882,
                            "status": "open"
                        },
                        {
                            "id": 19470,
                            "race_num": 4,
                            "description": "Ankunding Isle",
                            "outcome": 1499723947,
                            "status": "open"
                        },
                        {
                            "id": 83099,
                            "race_num": 5,
                            "description": "Charlene Extension",
                            "outcome": 1499724620,
                            "status": "open"
                        },
                        {
                            "id": 15210,
                            "race_num": 6,
                            "description": "Renner Locks",
                            "outcome": 1499724882,
                            "status": "open"
                        },
                        {
                            "id": 71212,
                            "race_num": 7,
                            "description": "Kilback Knoll",
                            "outcome": 1499725222,
                            "status": "open"
                        },
                        {
                            "id": 31952,
                            "race_num": 8,
                            "description": "Ida Loop",
                            "outcome": 1499725396,
                            "status": "open"
                        },
                        {
                            "id": 55381,
                            "race_num": 9,
                            "description": "Dayne Forks",
                            "outcome": 1499725422,
                            "status": "open"
                        },
                        {
                            "id": 10637,
                            "race_num": 10,
                            "description": "Stamm Key",
                            "outcome": 1499726080,
                            "status": "open"
                        }
                    ]
                },
                {
                    "id": 2765,
                    "name": "West Pearlie",
                    "type": "G",
                    "date": "2017-07-11",
                    "country": "AU",
                    "events": [
                        {
                            "id": 40048,
                            "race_num": 1,
                            "description": "Loy Shoals",
                            "outcome": 1499723071,
                            "status": "open"
                        },
                        {
                            "id": 37332,
                            "race_num": 2,
                            "description": "Emmerich Isle",
                            "outcome": 1499723079,
                            "status": "open"
                        },
                        {
                            "id": 33599,
                            "race_num": 3,
                            "description": "Anderson Canyon",
                            "outcome": 1499723080,
                            "status": "open"
                        },
                        {
                            "id": 46933,
                            "race_num": 4,
                            "description": "Brandy Club",
                            "outcome": 1499723175,
                            "status": "open"
                        },
                        {
                            "id": 61286,
                            "race_num": 5,
                            "description": "Hegmann Squares",
                            "outcome": 1499723186,
                            "status": "open"
                        },
                        {
                            "id": 26193,
                            "race_num": 6,
                            "description": "Heathcote Trafficway",
                            "outcome": 1499723386,
                            "status": "open"
                        },
                        {
                            "id": 70956,
                            "race_num": 7,
                            "description": "Hoeger Mission",
                            "outcome": 1499723394,
                            "status": "open"
                        },
                        {
                            "id": 49529,
                            "race_num": 8,
                            "description": "Ryan Loaf",
                            "outcome": 1499723404,
                            "status": "open"
                        },
                        {
                            "id": 79170,
                            "race_num": 9,
                            "description": "West Mews",
                            "outcome": 1499724493,
                            "status": "open"
                        },
                        {
                            "id": 75495,
                            "race_num": 10,
                            "description": "Stark Stream",
                            "outcome": 1499725871,
                            "status": "open"
                        }
                    ]
                },
                {
                    "id": 6364,
                    "name": "East Tyrese",
                    "type": "G",
                    "date": "2017-07-11",
                    "country": "AU",
                    "events": [
                        {
                            "id": 18691,
                            "race_num": 1,
                            "description": "Oberbrunner Locks",
                            "outcome": 1499723078,
                            "status": "open"
                        },
                        {
                            "id": 12367,
                            "race_num": 2,
                            "description": "Crooks Light",
                            "outcome": 1499723294,
                            "status": "open"
                        },
                        {
                            "id": 16445,
                            "race_num": 3,
                            "description": "Mitchell Land",
                            "outcome": 1499724273,
                            "status": "open"
                        },
                        {
                            "id": 32371,
                            "race_num": 4,
                            "description": "Jerald Dale",
                            "outcome": 1499724575,
                            "status": "open"
                        },
                        {
                            "id": 76303,
                            "race_num": 5,
                            "description": "Kohler Hills",
                            "outcome": 1499724578,
                            "status": "open"
                        },
                        {
                            "id": 71408,
                            "race_num": 6,
                            "description": "Kihn Mall",
                            "outcome": 1499725173,
                            "status": "open"
                        },
                        {
                            "id": 38569,
                            "race_num": 7,
                            "description": "Rutherford Brook",
                            "outcome": 1499725322,
                            "status": "open"
                        },
                        {
                            "id": 41382,
                            "race_num": 8,
                            "description": "Jacklyn Causeway",
                            "outcome": 1499725725,
                            "status": "open"
                        },
                        {
                            "id": 36177,
                            "race_num": 9,
                            "description": "Ezequiel Course",
                            "outcome": 1499726106,
                            "status": "open"
                        },
                        {
                            "id": 46964,
                            "race_num": 10,
                            "description": "Doyle Cove",
                            "outcome": 1499726111,
                            "status": "open"
                        }
                    ]
                },
                {
                    "id": 6328,
                    "name": "Venatown",
                    "type": "G",
                    "date": "2017-07-11",
                    "country": "AU",
                    "events": [
                        {
                            "id": 19272,
                            "race_num": 1,
                            "description": "Blanda Squares",
                            "outcome": 1499723208,
                            "status": "open"
                        },
                        {
                            "id": 73141,
                            "race_num": 2,
                            "description": "Joanne Inlet",
                            "outcome": 1499723369,
                            "status": "open"
                        },
                        {
                            "id": 85468,
                            "race_num": 3,
                            "description": "Douglas Plaza",
                            "outcome": 1499723525,
                            "status": "open"
                        },
                        {
                            "id": 53271,
                            "race_num": 4,
                            "description": "Kenneth Underpass",
                            "outcome": 1499723583,
                            "status": "open"
                        },
                        {
                            "id": 46179,
                            "race_num": 5,
                            "description": "Vivienne Shore",
                            "outcome": 1499723672,
                            "status": "open"
                        },
                        {
                            "id": 34416,
                            "race_num": 6,
                            "description": "Grayson Village",
                            "outcome": 1499724924,
                            "status": "open"
                        },
                        {
                            "id": 27486,
                            "race_num": 7,
                            "description": "Smith Vista",
                            "outcome": 1499724931,
                            "status": "open"
                        },
                        {
                            "id": 85146,
                            "race_num": 8,
                            "description": "Morissette Heights",
                            "outcome": 1499725401,
                            "status": "open"
                        },
                        {
                            "id": 46431,
                            "race_num": 9,
                            "description": "Macie Crossroad",
                            "outcome": 1499725416,
                            "status": "open"
                        },
                        {
                            "id": 60688,
                            "race_num": 10,
                            "description": "Schmidt Route",
                            "outcome": 1499726079,
                            "status": "open"
                        }
                    ]
                },
                {
                    "id": 1228,
                    "name": "West Orrinview",
                    "type": "G",
                    "date": "2017-07-10",
                    "country": "AU",
                    "events": [
                        {
                            "id": 31204,
                            "race_num": 1,
                            "description": "Bechtelar Wells",
                            "outcome": 1499723158,
                            "status": "open"
                        },
                        {
                            "id": 16749,
                            "race_num": 2,
                            "description": "Reichel Pines",
                            "outcome": 1499723459,
                            "status": "open"
                        },
                        {
                            "id": 28038,
                            "race_num": 3,
                            "description": "O'Hara Plains",
                            "outcome": 1499724196,
                            "status": "open"
                        },
                        {
                            "id": 69401,
                            "race_num": 4,
                            "description": "Cummerata Forks",
                            "outcome": 1499724240,
                            "status": "open"
                        },
                        {
                            "id": 55507,
                            "race_num": 5,
                            "description": "Ceasar Extension",
                            "outcome": 1499724410,
                            "status": "open"
                        },
                        {
                            "id": 68735,
                            "race_num": 6,
                            "description": "Arnold Ranch",
                            "outcome": 1499724829,
                            "status": "open"
                        },
                        {
                            "id": 62821,
                            "race_num": 7,
                            "description": "Loyal Loaf",
                            "outcome": 1499725240,
                            "status": "open"
                        },
                        {
                            "id": 49527,
                            "race_num": 8,
                            "description": "Hyman Way",
                            "outcome": 1499725443,
                            "status": "open"
                        },
                        {
                            "id": 19267,
                            "race_num": 9,
                            "description": "Leonard Canyon",
                            "outcome": 1499725988,
                            "status": "open"
                        },
                        {
                            "id": 29778,
                            "race_num": 10,
                            "description": "Sydney Rue",
                            "outcome": 1499726368,
                            "status": "open"
                        }
                    ]
                },
                {
                    "id": 1374,
                    "name": "West Salma",
                    "type": "G",
                    "date": "2017-07-11",
                    "country": "AU",
                    "events": [
                        {
                            "id": 12137,
                            "race_num": 1,
                            "description": "Erica Mountain",
                            "outcome": 1499723187,
                            "status": "open"
                        },
                        {
                            "id": 12642,
                            "race_num": 2,
                            "description": "Rudy Mountains",
                            "outcome": 1499723195,
                            "status": "open"
                        },
                        {
                            "id": 22727,
                            "race_num": 3,
                            "description": "Wiley Mountains",
                            "outcome": 1499723328,
                            "status": "open"
                        },
                        {
                            "id": 70015,
                            "race_num": 4,
                            "description": "Langworth Loaf",
                            "outcome": 1499723624,
                            "status": "open"
                        },
                        {
                            "id": 71788,
                            "race_num": 5,
                            "description": "Maude Cliffs",
                            "outcome": 1499723645,
                            "status": "open"
                        },
                        {
                            "id": 12895,
                            "race_num": 6,
                            "description": "Clifford Courts",
                            "outcome": 1499724487,
                            "status": "open"
                        },
                        {
                            "id": 34508,
                            "race_num": 7,
                            "description": "Zieme Landing",
                            "outcome": 1499724518,
                            "status": "open"
                        },
                        {
                            "id": 21511,
                            "race_num": 8,
                            "description": "Klocko Dam",
                            "outcome": 1499724871,
                            "status": "open"
                        },
                        {
                            "id": 39626,
                            "race_num": 9,
                            "description": "Marquise Loop",
                            "outcome": 1499725718,
                            "status": "open"
                        },
                        {
                            "id": 71457,
                            "race_num": 10,
                            "description": "Terence Rapids",
                            "outcome": 1499725788,
                            "status": "open"
                        }
                    ]
                },
                {
                    "id": 2841,
                    "name": "North Assuntaside",
                    "type": "H",
                    "date": "2017-07-11",
                    "country": "AU",
                    "events": [
                        {
                            "id": 34179,
                            "race_num": 1,
                            "description": "Jaleel Summit",
                            "outcome": 1499723151,
                            "status": "open"
                        },
                        {
                            "id": 26825,
                            "race_num": 2,
                            "description": "Dessie Ways",
                            "outcome": 1499723238,
                            "status": "open"
                        },
                        {
                            "id": 53156,
                            "race_num": 3,
                            "description": "Cruz Underpass",
                            "outcome": 1499724211,
                            "status": "open"
                        },
                        {
                            "id": 79492,
                            "race_num": 4,
                            "description": "Howe Via",
                            "outcome": 1499724236,
                            "status": "open"
                        },
                        {
                            "id": 70639,
                            "race_num": 5,
                            "description": "Towne Mountains",
                            "outcome": 1499724333,
                            "status": "open"
                        },
                        {
                            "id": 33931,
                            "race_num": 6,
                            "description": "DuBuque Square",
                            "outcome": 1499724375,
                            "status": "open"
                        },
                        {
                            "id": 79309,
                            "race_num": 7,
                            "description": "Senger Field",
                            "outcome": 1499725227,
                            "status": "open"
                        },
                        {
                            "id": 25995,
                            "race_num": 8,
                            "description": "McKenzie Walks",
                            "outcome": 1499725405,
                            "status": "open"
                        },
                        {
                            "id": 56975,
                            "race_num": 9,
                            "description": "Lowe Mountains",
                            "outcome": 1499725561,
                            "status": "open"
                        },
                        {
                            "id": 55178,
                            "race_num": 10,
                            "description": "Stoltenberg Stravenue",
                            "outcome": 1499726466,
                            "status": "open"
                        }
                    ]
                },
                {
                    "id": 3604,
                    "name": "Gerholdton",
                    "type": "H",
                    "date": "2017-07-11",
                    "country": "AU",
                    "events": [
                        {
                            "id": 73798,
                            "race_num": 1,
                            "description": "Maryse Key",
                            "outcome": 1499723594,
                            "status": "open"
                        },
                        {
                            "id": 35365,
                            "race_num": 2,
                            "description": "Paxton Manors",
                            "outcome": 1499723837,
                            "status": "open"
                        },
                        {
                            "id": 69394,
                            "race_num": 3,
                            "description": "Crist Greens",
                            "outcome": 1499723952,
                            "status": "open"
                        },
                        {
                            "id": 16575,
                            "race_num": 4,
                            "description": "Damien Trafficway",
                            "outcome": 1499724122,
                            "status": "open"
                        },
                        {
                            "id": 64861,
                            "race_num": 5,
                            "description": "Matt Club",
                            "outcome": 1499724127,
                            "status": "open"
                        },
                        {
                            "id": 75912,
                            "race_num": 6,
                            "description": "Kirlin Fort",
                            "outcome": 1499724278,
                            "status": "open"
                        },
                        {
                            "id": 75483,
                            "race_num": 7,
                            "description": "Kozey Extensions",
                            "outcome": 1499725430,
                            "status": "open"
                        },
                        {
                            "id": 15962,
                            "race_num": 8,
                            "description": "Cielo Landing",
                            "outcome": 1499725704,
                            "status": "open"
                        },
                        {
                            "id": 79536,
                            "race_num": 9,
                            "description": "Jaeden Divide",
                            "outcome": 1499725752,
                            "status": "open"
                        },
                        {
                            "id": 84569,
                            "race_num": 10,
                            "description": "Romaguera Park",
                            "outcome": 1499726545,
                            "status": "open"
                        }
                    ]
                },
                {
                    "id": 3824,
                    "name": "New Matteo",
                    "type": "H",
                    "date": "2017-07-11",
                    "country": "AU",
                    "events": [
                        {
                            "id": 62555,
                            "race_num": 1,
                            "description": "Lavada Ridges",
                            "outcome": 1499723837,
                            "status": "open"
                        },
                        {
                            "id": 43125,
                            "race_num": 2,
                            "description": "Schamberger Lights",
                            "outcome": 1499724558,
                            "status": "open"
                        },
                        {
                            "id": 85012,
                            "race_num": 3,
                            "description": "Mary Pass",
                            "outcome": 1499724659,
                            "status": "open"
                        },
                        {
                            "id": 58427,
                            "race_num": 4,
                            "description": "Tom Point",
                            "outcome": 1499724675,
                            "status": "open"
                        },
                        {
                            "id": 60389,
                            "race_num": 5,
                            "description": "Ericka Mount",
                            "outcome": 1499724769,
                            "status": "open"
                        },
                        {
                            "id": 74324,
                            "race_num": 6,
                            "description": "Lakin Motorway",
                            "outcome": 1499725754,
                            "status": "open"
                        },
                        {
                            "id": 88971,
                            "race_num": 7,
                            "description": "Lockman Crest",
                            "outcome": 1499725890,
                            "status": "open"
                        },
                        {
                            "id": 77134,
                            "race_num": 8,
                            "description": "Darrion Summit",
                            "outcome": 1499725977,
                            "status": "open"
                        },
                        {
                            "id": 18944,
                            "race_num": 9,
                            "description": "Wilkinson Greens",
                            "outcome": 1499726029,
                            "status": "open"
                        },
                        {
                            "id": 11615,
                            "race_num": 10,
                            "description": "Walsh Green",
                            "outcome": 1499726193,
                            "status": "open"
                        }
                    ]
                },
                {
                    "id": 7482,
                    "name": "New Alejandraside",
                    "type": "H",
                    "date": "2017-07-11",
                    "country": "AU",
                    "events": [
                        {
                            "id": 43305,
                            "race_num": 1,
                            "description": "Murray Glens",
                            "outcome": 1499723084,
                            "status": "open"
                        },
                        {
                            "id": 38656,
                            "race_num": 2,
                            "description": "Kathleen Rue",
                            "outcome": 1499723700,
                            "status": "open"
                        },
                        {
                            "id": 31571,
                            "race_num": 3,
                            "description": "Caden Pass",
                            "outcome": 1499723795,
                            "status": "open"
                        },
                        {
                            "id": 31509,
                            "race_num": 4,
                            "description": "Edgardo Parkway",
                            "outcome": 1499725258,
                            "status": "open"
                        },
                        {
                            "id": 85189,
                            "race_num": 5,
                            "description": "Bernier Junction",
                            "outcome": 1499725431,
                            "status": "open"
                        },
                        {
                            "id": 37633,
                            "race_num": 6,
                            "description": "Grant Road",
                            "outcome": 1499725432,
                            "status": "open"
                        },
                        {
                            "id": 22362,
                            "race_num": 7,
                            "description": "Emery Avenue",
                            "outcome": 1499725910,
                            "status": "open"
                        },
                        {
                            "id": 49308,
                            "race_num": 8,
                            "description": "Sincere Walks",
                            "outcome": 1499726085,
                            "status": "open"
                        },
                        {
                            "id": 67377,
                            "race_num": 9,
                            "description": "Langworth View",
                            "outcome": 1499726287,
                            "status": "open"
                        },
                        {
                            "id": 51807,
                            "race_num": 10,
                            "description": "Crooks Rapid",
                            "outcome": 1499726354,
                            "status": "open"
                        }
                    ]
                }
            ],
            "hash": ""
        },
        {
            "type": "competitor",
            "data": {
                "event_id": "30395",
                "competitors": [
                    {
                        "id": 708751,
                        "name": "Prof. Raymond Tremblay III",
                        "saddle_number": 1,
                        "barrier": 2,
                        "eliminated": "no",
                        "apn": "2.2,2.4,3.6,5.7,6.6,8.2,9.2,9.4",
                        "fixed_win_price": "12.23",
                        "fixed_place_price": "30.86"
                    },
                    {
                        "id": 827690,
                        "name": "Abdiel Ryan",
                        "saddle_number": 2,
                        "barrier": 3,
                        "eliminated": "yes",
                        "apn": "1.8,2.2,2.3,2.4,2.5,5,8.4,9.2",
                        "fixed_win_price": "23.28",
                        "fixed_place_price": "57.39"
                    },
                    {
                        "id": 682372,
                        "name": "Mr. Lorenza Welch Jr.",
                        "saddle_number": 3,
                        "barrier": 4,
                        "eliminated": "yes",
                        "apn": "2.2,2.3,3.6,4.2,4.4,5.6,5.6,6.5",
                        "fixed_win_price": "42.07",
                        "fixed_place_price": "92.47"
                    },
                    {
                        "id": 542769,
                        "name": "Kolby Ziemann MD",
                        "saddle_number": 4,
                        "barrier": 5,
                        "eliminated": "no",
                        "apn": "1.5,5.3,5.6,7,7.4,8,8.3,9",
                        "fixed_win_price": "49.48",
                        "fixed_place_price": "90.5"
                    },
                    {
                        "id": 394478,
                        "name": "Dr. Jeremy Sipes MD",
                        "saddle_number": 5,
                        "barrier": 6,
                        "eliminated": "no",
                        "apn": "1.9,2.2,4,5.4,5.6,7.5,9.8,9.9",
                        "fixed_win_price": "3.36",
                        "fixed_place_price": "81.26"
                    },
                    {
                        "id": 874992,
                        "name": "Thad Reichert",
                        "saddle_number": 6,
                        "barrier": 7,
                        "eliminated": "yes",
                        "apn": "2.5,3.3,3.3,4.7,4.7,5.7,7.4,8",
                        "fixed_win_price": "76.84",
                        "fixed_place_price": "71.66"
                    },
                    {
                        "id": 721910,
                        "name": "Dr. Jairo Heidenreich II",
                        "saddle_number": 7,
                        "barrier": 8,
                        "eliminated": "yes",
                        "apn": "1.5,3.5,4,6.8,6.9,7.3,8.7,9.1",
                        "fixed_win_price": "82.67",
                        "fixed_place_price": "91"
                    },
                    {
                        "id": 817362,
                        "name": "Mr. Kenny Yost",
                        "saddle_number": 8,
                        "barrier": 9,
                        "eliminated": "no",
                        "apn": "1.6,3.5,5,5.7,6.7,7.2,7.6,8.6",
                        "fixed_win_price": "99.77",
                        "fixed_place_price": "85.09"
                    },
                    {
                        "id": 832153,
                        "name": "Paul Schamberger",
                        "saddle_number": 9,
                        "barrier": 10,
                        "eliminated": "no",
                        "apn": "1.5,2.8,3,3.6,4.8,4.9,5.7,5.9",
                        "fixed_win_price": "3.61",
                        "fixed_place_price": "37.61"
                    },
                    {
                        "id": 675117,
                        "name": "Miss Judy Baumbach Jr.",
                        "saddle_number": 10,
                        "barrier": 11,
                        "eliminated": "yes",
                        "apn": "2.3,2.6,3.2,3.4,5.8,7,7.8,8.2",
                        "fixed_win_price": "10",
                        "fixed_place_price": "41.94"
                    }
                ],
                "event": [
                    {
                        "id": 30395,
                        "race_num": 1,
                        "description": "Jimmie Islands",
                        "outcome": 1499722976,
                        "status": "open"
                    }
                ]
            },
            "hash": ""
        }
    ]
}