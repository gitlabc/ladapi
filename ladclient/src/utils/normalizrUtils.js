
import { normalize, schema } from 'normalizr';

export function getNext5MeetingsAndEvents(originalData) {
    const eventSchema = new schema.Entity('events', {}, {
        processStrategy: (value, parent, key) => {
            return { ...value, meeting_id: parent.id, type: parent.type };
        }
    });
    const meetingSchema = new schema.Entity('meetings', {
        events: [eventSchema],
    });
    const normalizedData = normalize(originalData, [meetingSchema]);
    return normalizedData;
}

export function getNext5CompetitorData(json){
    let competitorData = json.updates.filter(update => {
        return update.type === 'competitor' && update.data.competitors.length > 0;
    });
    return competitorData && competitorData[0].data;
}
