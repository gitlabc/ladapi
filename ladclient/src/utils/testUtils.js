import * as meetingsPackage from './meetings.json';
import { getCurrentTimestamp } from './commonUtils';
import { initRaceTypeState } from '../redux/reducers/initialState';
import { getNext5MeetingsAndEvents, getNext5CompetitorData } from './normalizrUtils';

const meetingsJson = meetingsPackage.updates[0].data;
const objEvents = getNext5MeetingsAndEvents(meetingsJson).entities.events;
const objMeetings = getNext5MeetingsAndEvents(meetingsJson).entities.meetings;
const { competitors } = getNext5CompetitorData(meetingsPackage);

const meetings = Object.keys(objMeetings).map(key => (objMeetings[key]));
const event = objEvents[Object.keys(objEvents)[0]];
const meeting = objMeetings[event.meeting_id];
const meetingEvents = (meeting.events).map(eventId => {
    return objEvents[eventId + ''];
}) || [];

const competitor = competitors[0];
const now = getCurrentTimestamp();

const top5Events = Object.keys(objEvents).slice(0, 5).map(key => {
    let obj = objEvents[key];
    obj.meetingName = objMeetings[obj.meeting_id].name;
    return obj;
});

const raceTypes = initRaceTypeState.toJS();

export { event, meetingEvents, competitors, meeting, meetings, competitor, now, top5Events, raceTypes };
