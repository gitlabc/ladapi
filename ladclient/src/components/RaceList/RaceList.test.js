import React from 'react';
import renderer from 'react-test-renderer';
import RaceList from './RaceList';
import { shallow, mount, render } from 'enzyme';

import { top5Events } from '../../utils/testUtils';

describe('RaceList', () => {
    test('should generate the RaceList', () => {
        const wrapper = renderer.create(<RaceList
            top5Events={top5Events}
            onRaceClick={() => { }}
        />);

        expect(wrapper).toMatchSnapshot();
    });


    test('should have 5 RaceListItem', () => {
        const wrapper = shallow(
            <RaceList
                top5Events={top5Events}
                onRaceClick={() => { }}
            />
        );
        expect(wrapper.find('RaceListItem')).toHaveLength(5);
    });

    test('should have 5 buttons', () => {
        const wrapper = render(
            <RaceList
                top5Events={top5Events}
                onRaceClick={() => { }}
            />
        );
        expect(wrapper.find('button')).toHaveLength(5);
    });

    test('simulate click events', () => {
        let eventId = 0;
        let meetingId = 0;
        const onEventClick = jest.fn().mockImplementation((e, m) => () => {
            eventId = e;
            meetingId = m;
        });
        const wrapper = mount(
            <RaceList
                top5Events={top5Events}
                onRaceClick={onEventClick}
            />
        );
        const index = 1;
        expect(onEventClick.mock.calls).toHaveLength(5);
        wrapper.find('button').forEach((btn, i) => {
            if (i === index) {
                btn.simulate('click');
            }
        });
        expect([top5Events[index].id, top5Events[index].meeting_id]).toEqual([eventId, meetingId]);
    });
});
