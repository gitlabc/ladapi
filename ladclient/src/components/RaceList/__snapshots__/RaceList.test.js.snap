// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaceList should generate the RaceList 1`] = `
<div
  className={undefined}
>
  <div
    className="ant-row-flex ant-row-flex-middle"
    onClick={undefined}
    style={undefined}
  >
    <div
      className="ant-col-xs-2 ant-col-sm-1"
    >
      <div
        className=""
      />
       
    </div>
    <div
      className="ant-col-xs-14 ant-col-sm-19"
    >
      NEW CYRUS
    </div>
    <div
      className="ant-col-xs-2 ant-col-sm-1"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-sm"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <span>
          R
        </span>
        10
      </button>
    </div>
    <div
      className="ant-col-xs-4 ant-col-sm-2"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
        
    </div>
    <div
      className="ant-col-xs-2 ant-col-sm-1"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <i
        className="anticon anticon-right"
      />
    </div>
  </div>
  <div
    className="ant-row-flex ant-row-flex-middle"
    onClick={undefined}
    style={undefined}
  >
    <div
      className="ant-col-xs-2 ant-col-sm-1"
    >
      <div
        className=""
      />
       
    </div>
    <div
      className="ant-col-xs-14 ant-col-sm-19"
    >
      NEW MATTEO
    </div>
    <div
      className="ant-col-xs-2 ant-col-sm-1"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-sm"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <span>
          R
        </span>
        10
      </button>
    </div>
    <div
      className="ant-col-xs-4 ant-col-sm-2"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
        
    </div>
    <div
      className="ant-col-xs-2 ant-col-sm-1"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <i
        className="anticon anticon-right"
      />
    </div>
  </div>
  <div
    className="ant-row-flex ant-row-flex-middle"
    onClick={undefined}
    style={undefined}
  >
    <div
      className="ant-col-xs-2 ant-col-sm-1"
    >
      <div
        className=""
      />
       
    </div>
    <div
      className="ant-col-xs-14 ant-col-sm-19"
    >
      WEST SALMA
    </div>
    <div
      className="ant-col-xs-2 ant-col-sm-1"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-sm"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <span>
          R
        </span>
        1
      </button>
    </div>
    <div
      className="ant-col-xs-4 ant-col-sm-2"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
        
    </div>
    <div
      className="ant-col-xs-2 ant-col-sm-1"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <i
        className="anticon anticon-right"
      />
    </div>
  </div>
  <div
    className="ant-row-flex ant-row-flex-middle"
    onClick={undefined}
    style={undefined}
  >
    <div
      className="ant-col-xs-2 ant-col-sm-1"
    >
      <div
        className=""
      />
       
    </div>
    <div
      className="ant-col-xs-14 ant-col-sm-19"
    >
      EAST TYRESE
    </div>
    <div
      className="ant-col-xs-2 ant-col-sm-1"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-sm"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <span>
          R
        </span>
        2
      </button>
    </div>
    <div
      className="ant-col-xs-4 ant-col-sm-2"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
        
    </div>
    <div
      className="ant-col-xs-2 ant-col-sm-1"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <i
        className="anticon anticon-right"
      />
    </div>
  </div>
  <div
    className="ant-row-flex ant-row-flex-middle"
    onClick={undefined}
    style={undefined}
  >
    <div
      className="ant-col-xs-2 ant-col-sm-1"
    >
      <div
        className=""
      />
       
    </div>
    <div
      className="ant-col-xs-14 ant-col-sm-19"
    >
      WEST SALMA
    </div>
    <div
      className="ant-col-xs-2 ant-col-sm-1"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-sm"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <span>
          R
        </span>
        2
      </button>
    </div>
    <div
      className="ant-col-xs-4 ant-col-sm-2"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
        
    </div>
    <div
      className="ant-col-xs-2 ant-col-sm-1"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <i
        className="anticon anticon-right"
      />
    </div>
  </div>
</div>
`;
